src/Service.Tests/CosmosTests/SchemaGeneratorTest.cs

```diff
@@ -0,0 +1,288 @@
+  id : ID!,
+  name : String!,
+  price : Float!,
+  inStock : Boolean!,
+  tags : [String]!,
+  dimensions : Dimensions!,
+  manufacturedDate : Date!,
+  relatedProducts : [Int]!
+}
+type Dimensions {
+  length : Float!,
+  width : Float!,
+  height : Float!
+}";
+
+            AreEqualAfterCleanup(expectedSchema, gqlSchema);
+        }
+
+        /// <summary>
+        /// Tests the <see cref="SchemaGenerator.Generate"/> method with a complex JSON object containing nested objects and arrays.
+        /// Ensures that the generated GraphQL schema accurately represents the nested structure and data types.
+        /// </summary>
+        [TestMethod]
+        public void TestComplexJsonObject()
+        {
+            List<JsonDocument> jsonArray = new() {
+                JsonDocument.Parse(@"{
+                  ""name"": ""John Doe"",
+                  ""age"": 30,
+                  ""address"": {
+                    ""street"": ""123 Main St"",
+                    ""city"": ""Anytown"",
+                    ""state"": ""CA"",
+                    ""zip"": ""12345"",
+                    ""coordinates"": {
+                      ""latitude"": 34.0522,
+                      ""longitude"": -118.2437
+                    }
+                  },
+                  ""emails"": [
+                    ""john.doe@example.com"",
+                    ""john.doe@work.com""
+                  ],
+                  ""phoneNumbers"": [
+                    {
+                      ""type"": ""home"",
+                      ""number"": ""555-555-5555""
+                    },
+                    {
+                      ""type"": ""work"",
+                      ""number"": ""555-555-5556""
+                    }
+                  ]
+                }")};
+
+            string gqlSchema = SchemaGenerator.Generate(jsonArray, "containerName");
+
+            string expectedSchema = @"type ContainerName @model {
+  name : String!,
+  age : Int!,
+  address : Address!,
+  emails : [String]!,
+  phoneNumbers : [PhoneNumber]!
+}
+type Address {
+  street : String!,
+  city : String!,
+  state : String!,
+  zip : String!,
+  coordinates : Coordinates!
+}
+type Coordinates {
+  latitude : Float!,
+  longitude : Float!
+}
+type PhoneNumber {
+  type : String!,
+  number : String!
+}";
+
+            AreEqualAfterCleanup(expectedSchema, gqlSchema);
+        }
+
+        /// <summary>
+        /// Tests the <see cref="SchemaGenerator.Generate"/> method with a JSON array containing mixed data types.
+        /// Verifies that the generated GraphQL schema includes all possible fields and types present in the input data.
+        /// </summary>
+        [TestMethod]
+        public void TestMixedJsonArray()
+        {
+            List<JsonDocument> jsonArray = new() {
+                JsonDocument.Parse(@"{ ""name"": ""John"", ""age"": 30, ""isStudent"": false, ""birthDate"": ""1980-01-01T00:00:00Z"" }"),
```

Comment:
what happens when birthday is null? i recognized in other code that there is a case for attempting to parse a null value as a date.

Reply:
it won't fail, it will go to else and consider it as `string`.